pool/paymentmgr_test.go

```diff
@@ -98,11 +187,168 @@ func testPaymentMgr(t *testing.T, db *bolt.DB) {
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	minimumTimeBytes := nanoToBigEndianBytes(minimumTime)
+	maximumTimeBytes := nanoToBigEndianBytes(maximumTime)
+
+	// Fetch eligible shares using the minimum and maximum time range.
+	shares, err := mgr.PPSEligibleShares(minimumTimeBytes, maximumTimeBytes)
+	if err != nil {
+		t.Fatalf("PPSEligibleShares: unexpected error: %v", err)
+	}
+
+	// Ensure the returned share count is as expected.
+	if len(shares) != expectedShareCount {
+		t.Fatalf("PPS error: expected %v eligible PPS shares, got %v",
+			expectedShareCount, len(shares))
+	}
+
+	forAccX := 0
+	forAccY := 0
+	for _, share := range shares {
+		if share.Account == xID {
+			forAccX++
+		}
+
+		if share.Account == yID {
+			forAccY++
+		}
+	}
+
+	// Ensure account x and account y both have shares returned.
+	if forAccX == 0 || forAccY == 0 {
+		t.Fatalf("PPS error: expected shares for account X and Y, "+
+			"got %v (for x), %v (for y).", forAccX, forAccY)
+	}
+
+	// Ensure account x and account y have equal number of shares.
+	if forAccX != forAccY {
+		t.Fatalf("PPS error: expected equal shares for account X and Y, "+
+			"got %v (for x), %v (for y).", forAccX, forAccY)
+	}
+
+	// Ensure account x and account y have shares equal to the share count.
+	if forAccX != shareCount || forAccY != shareCount {
+		t.Fatalf("PPS error: expected share counts of %v for account X and Y, "+
+			"got %v (for x), %v (for y).", shareCount, forAccX, forAccY)
+	}
+
+	// Empty the share bucket.
+	err = emptyBucket(db, shareBkt)
+	if err != nil {
+		t.Fatalf("emptyBucket error: %v", err)
+	}
+
+	// Create a share below the minimum exclusive PPLNS time for account x.
+	err = persistShare(db, xID, weight, belowMinimumTime)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create a share below the minimum exxcuive PPLNS time for account y.
+	err = persistShare(db, yID, weight, belowMinimumTime)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create a share at minimum exclusive PPLNS time for account x.
+	err = persistShare(db, xID, weight, minimumTime)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create a share at minimum exclusive PPLNS time for account y.
+	err = persistShare(db, yID, weight, minimumTime)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create a share above minimum exclusive PPLNS time for account x.
+	err = persistShare(db, xID, weight, maximumTime)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create a share above minimum exclusive PPLNS time for account y.
+	err = persistShare(db, yID, weight, aboveMaximumTime)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	shares, err = mgr.PPLNSEligibleShares(minimumTimeBytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Ensure the returned number of shates is as expected.
+	if len(shares) != expectedShareCount {
+		t.Fatalf("PPLNS error: expected %v eligible PPLNS shares, got %v",
+			expectedShareCount, len(shares))
+	}
+
+	// Ensure account x and account y both have shares returned.
+	if forAccX == 0 || forAccY == 0 {
```

Comment:
`forAccX` and `forAccY` have not been recalculated. This is still using the values from the previous test